python/thunder/rdds/series.py

```diff
@@ -56,6 +56,23 @@ def index(self):
         if self._index is None:
             self.populateParamsFromFirstRecord()
         return self._index
+        
+    @index.setter
+    def index(self, value):
+        try:
+            value[0]
+        except:
+            raise AttributeError("Index must be a non-empty indexable object")
+        if self._index is None:
```

Comment:
This shouldn't be necessary if, in line 72, you use `self.index` instead of `self._index`, because the automatic population from the first record will happen there.